backend/controller/cronjobs/cronjobs.go

```diff
@@ -153,344 +70,131 @@ func (s *Service) NewCronJobsForModule(ctx context.Context, module *schemapb.Mod
 				Schedule:      cronStr,
 				StartTime:     start,
 				NextExecution: next,
-				State:         model.CronJobStateIdle,
 				// DeploymentKey: Filled in by DAL
 			})
 		}
 	}
+	logger.Tracef("Found %d cron jobs", len(newJobs))
 	if len(merr) > 0 {
 		return nil, errors.Join(merr...)
 	}
 	return newJobs, nil
 }
 
-// CreatedOrReplacedDeloyment is only called by the responsible controller to its cron service, and will not be received by the other cron services.
-// When a controller creates/replaces a deployment, its cron job service is responsible for
-// the newly created cron jobs until other controllers have a chance to resync their list of jobs and start sharing responsibility of the new cron jobs.
-func (s *Service) CreatedOrReplacedDeloyment(ctx context.Context, newDeploymentKey model.DeploymentKey) {
-	// Rather than finding old/new cron jobs and updating our state, we can just resync the list of jobs
-	_ = s.syncJobsWithNewDeploymentKey(ctx, optional.Some(newDeploymentKey)) //nolint:errcheck // TODO(matt2e) is this valid?
-}
-
-// SyncJobs is run periodically via a scheduled task
-func (s *Service) syncJobs(ctx context.Context) (time.Duration, error) {
-	err := s.syncJobsWithNewDeploymentKey(ctx, optional.None[model.DeploymentKey]())
+// CreatedOrReplacedDeloyment is called by the responsible controller to its cron service, we can
+// schedule all cron jobs here since the cron_jobs rows are locked within the transaction and the
+// controllers won't step on each other.
+func (s *Service) CreatedOrReplacedDeloyment(ctx context.Context) {
```

Comment:
There was a chicken and egg problem using a singleton in the constructor to schedule all cron jobs, since all weren't available yet. Instead, we attempt to schedule all cron jobs whenever any deployment is replaced, and avoid the controller race condition by locking the corresponding `cron_job` rows in postgres.

Not ecstatic about this solution, open to ideas..